File: src/omnia/data/structure/mutable/MutableList.java
package omnia.data.structure.mutable;

import omnia.data.structure.List;

public interface MutableList<E> extends List<E>, MutableCollection<E> {

  void insertAt(int index, E element);

  E removeAt(int index);

  E replaceAt(int index, E element);

  static <E> MutableList<E> masking(java.util.List<E> javaList) {
    return new MaskingList<>(javaList);
  }
}

File: src/omnia/data/structure/mutable/MutableList.kt
package omnia.data.structure.mutable

import omnia.data.structure.List

interface MutableList<E> : List<E>, MutableCollection<E> {

  fun insertAt(index: Int, item: E)
  fun removeAt(index: Int): E
  fun replaceAt(index: Int, item: E): E

  companion object {

    fun <E> masking(javaList: kotlin.collections.MutableList<E>): MutableList<E> {
      return MaskingList(javaList)
    }
  }
}